Modules/hpe3par_snapshot.py

```diff
@@ -435,13 +468,188 @@ def restore_snapshot_online(
+           return (False, False, "Volume does not Exist", {})
+        
+        if retentionHours > expirationHours:
+           return (False, False, "Expiration time must be greater than or equal to retention time", {})   
+       
+        if task_freq_custom:          
+          if ' ' in task_freq_custom:            
+           taskcm = str(task_freq_custom).split()
+           if len(taskcm) == 5:
+               minutes_t = taskcm[0]               
+               hour_t = taskcm[1]
+               day_t = taskcm[2]
+               month_t = taskcm[3]  
+               dow_t = taskcm[4]
+               if '*' not in minutes_t:
+                  if (int(minutes_t) > 50 or int(minutes_t) < 0):
+                    return (False, False, "Invalid task freq", {})
+               else:
+                  if (len(minutes_t)>1 or minutes_t == ""):
+                    return (False, False, "Invalid task freq", {})
+               if '*' not in hour_t:
+                  if (int(hour_t) > 23 or int(hour_t) < 0):
+                    return (False, False, "Invalid task freq", {})
+               else:
+                  if (len(hour_t)>1 or hour_t == ""):
+                    return (False, False, "Invalid task freq", {})
+               if '*' not in day_t:
+                  if (int(day_t) > 31 or int(day_t) < 1):
+                    return (False, False, "Invalid task freq", {})
+               else:
+                  if (len(day_t)>1 or day_t == ""):
+                    return (False, False, "Invalid task freq", {})
+               if '*' not in month_t:
+                  if (int(month_t) > 12 or int(month_t) < 1):
+                    return (False, False, "Invalid task freq", {})
+               else:
+                  if (len(month_t)>1 or month_t == ""):
+                    return (False, False, "Invalid task freq", {})
+               if '*' not in dow_t:
+                  if (int(dow_t) > 6 or int(dow_t) < 0):
+                    return (False, False, "Invalid task freq", {})
+               else:
+                  if (len(dow_t)>1 or dow_t == ""):
+                    return (False, False, "Invalid task freq", {})
+           else:
+              return (False, False, "Invalid task freq", {})
+          else:
+              return (False, False, "Invalid task freq", {})
+          freq = task_freq_custom
+
+        if task_freq_custom and task_freq:
+            return (False, False, "Please Enter either task_freq or task_freq_custom", {})    
+        if not client_obj.scheduleExists(schedule_name):
+           cmd = ["createsv"]
+           if read_only:
+              cmd.append("-ro")
+           if expirationHours != None:
+              cmd.append("-exp")
+              cmd.append(str(expirationHours)+"h")
+           if retentionHours != None:
+                 cmd.append("-retain")
+                 cmd.append(str(retentionHours)+"h")
+           snap_string = ".@y@@m@@d@@H@@M@@S@"
```

Comment:
What is the intent behind naming the snapshot like this?

Reply:
it should be unique for each snapshot volume

Reply:
How will the user know the snapshot names? By looking at the code? That is not right. The user must have a choice to provide the naming convention, in my opinion

Also, if we are appending timestamp, we should be appending to the base volume name and not snapshot name. This is just an unnecessary addition

Reply:
Need clarification on snapshot name.